test/daos/job.spec.js

```diff
@@ -64,6 +64,17 @@ describe('Dao: Job', function () {
 
       assert.equal(jobData.id, 1);
     });
+    it('responds null when db returns undefined', async function () {
+      const mockJobStore = {
+        fetchJobById: sinon.stub().returns(undefined),
```

Comment:
fetchJobById is asynchronous and returns a promise, so you should switch this line to:
`fetchJobById: sinon.stub().resolves(undefined),`

Reply:
Done